### ISSUE TITLE
Sipml5 have issue For Hold/Resume in Chrome Last Version And FireFox last Version

### ISSUE THREAD

**Issue by majidbigdeli** [DoubangoTelecom/sipml5#323]
Sipml5 have issue For Hold/Resume in Chrome Last Version And FireFox last Version;
IN FireFox Last version  Call Terminated

**Comment by Udirley** [comment]
Hello

In my case the following occurs ... I put in the standby and I retrieve it once, putting on hold the second time I can not retrieve the call. I appreciate you helping me. Test done in the latest chrome.

**Comment by Udirley** [comment]
Hello

In my case the following occurs ... I put in the standby and I retrieve it once, putting on hold the second time I can not retrieve the call. I appreciate you helping me. Test done in the latest chrome.

Using the latest version of mozilla the call terminates when trying to put it on hold.
I am using asterisk 16.0.1.

In short ... In Chrome I can put and retrieve the wait only once, in Mozilla the call is ended on the first try.

**Comment by dlethin** [comment]
Your problem might be related to chrome entering into its next phase in the migration to the WebRTC unified plan.  Google now implements the Unified-Plan by default for WebRTC in Chrome 72.  See [here](https://www.callstats.io/blog/what-is-unified-plan-and-how-will-it-affect-your-webrtc-development) for details.  There is a link to a transition guide.  It might be that specifying you want to use Plan-B will resolve your problem -- at least temporarily until the next phase when Google removes Plan-b implementation entirely.

**Comment by majidbigdeli** [comment]
@dlethin  thank you . Issue Fixed With Plan-B

**Comment by pr0c** [comment]
@majidbigdeli i have the same issue. Can you help me to solve it? How i can use "Plan-B"?

**Comment by majidbigdeli** [comment]
@Udirley You Can Fix Issue With Change In tmedia_session_jsep.js File .
Please open tmedia_session_jsep.js File And Then Find  new window.RTCPeerConnection
And Then Change 
From

`this.o_pc = new window.RTCPeerConnection(
                (o_iceServers && !o_iceServers.length) ? null : { iceServers: o_iceServers, rtcpMuxPolicy: "negotiate" }, // empty array is used to disable STUN/TURN.
                this.o_media_constraints
        );`

To 

`this.o_pc = new window.RTCPeerConnection(
			(o_iceServers && !o_iceServers.length) ? null : { iceServers: o_iceServers, rtcpMuxPolicy: "negotiate", sdpSemantics: "plan-b" }, // empty array is used to disable STUN/TURN.
                this.o_media_constraints
        );`

**Comment by majidbigdeli** [comment]
@Udirley 
To improve performance
In tmedia_session_jsep.js File 
Find tmedia_session_jsep01.onIceCandidate Then Change 

From

``` 
tmedia_session_jsep01.onIceCandidate = function (o_event, _This) {
    var This = (tmedia_session_jsep01.mozThis || _This);
    if (!This || !This.o_pc) {
        tsk_utils_log_error("This/PeerConnection is null: unexpected");
        return;
    }
    var iceState = (This.o_pc.iceGatheringState || This.o_pc.iceState);

    tsk_utils_log_info("onIceCandidate = " + iceState);

    if (iceState === "complete" || (o_event && !o_event.candidate)) {
        tsk_utils_log_info("ICE GATHERING COMPLETED!");
        tmedia_session_jsep01.onIceGatheringCompleted(This);
    }
    else if (This.o_pc.iceState === "failed") {
        tsk_utils_log_error("Ice state is 'failed'");
        This.o_mgr.callback(tmedia_session_events_e.GET_LO_FAILED, This.e_type);
    }
}
```

To

```
tmedia_session_jsep01.onIceCandidate = function (o_event, _This) {
	var This = (tmedia_session_jsep01.mozThis || _This);
	if (!This || !This.o_pc) {
		tsk_utils_log_error("This/PeerConnection is null: unexpected");
		return;
	}
	var iceState = (This.o_pc.iceGatheringState || This.o_pc.iceState);

	tsk_utils_log_info("onIceCandidate = " + iceState);

	if (!tmedia_session_jsep01.ice_timer) {
		tmedia_session_jsep01.ice_timer = setTimeout(function () {
			console.log("ICE timer...");
			tmedia_session_jsep01.onIceGatheringCompleted(This);
			tmedia_session_jsep01.ice_timer = null;
		}, 0);
	};

	if (iceState === "complete" || (o_event && !o_event.candidate)) {
		tsk_utils_log_info("ICE GATHERING COMPLETED!");
		tmedia_session_jsep01.onIceGatheringCompleted(This);
	}
	else if (This.o_pc.iceState === "failed") {
		tsk_utils_log_error("Ice state is 'failed'");
		This.o_mgr.callback(tmedia_session_events_e.GET_LO_FAILED, This.e_type);
	}
}
```

**Comment by majidbigdeli** [comment]
@pr0c

**Comment by dlethin** [comment]
The fix @majidbigdeli mentioned is what we did as a short term fix.   However, long term it will not be enough when Google releases a newer version of Chrome with the final phase of there unified-plan migration to remove support for plan-b entirely.   Hopefully someone with a bit more application knowledge might be able to offer insight as to what if anything would need to change to fully support the unified-plan in sipml5.  Another possibility might be that its the system the sipml5 application is connecting with that will need to change.

**Comment by albertius661** [comment]
Hello,
Chrome beta (Versión 95.0.4638.69 (Build oficial) (64 bits))no longer supports plan-b.
I was not sure if I should update my pbx for the unified plan to work. After testing sip.js, I have seen that it works fine and it is a sipml5 problem.
Anyone know any solution ??
Thanks.

**Comment by vikashkeshri14** [comment]
Is anyone fixed this issue
